### ISSUE TITLE
Clear example of bounded optimization?

### ISSUE THREAD

**Issue by microprediction** [facebookresearch/nevergrad#1019]
Hi, can you please complete the sentence for me? 

    `instrument = ng.p.Instrumentation(ng.p.Array(shape=(n_dim,)).bounds( ... something .. )

A clear example of bounded optim in the doc, hello world example would help (me, anyway) as this syntax is not the most intuitive.

**Comment by microprediction** [comment]
Okay, I guess it is:
`
 instrument = ng.p.Array(shape=(n_dim,)).set_bounds(lower=0.0,upper=1.0)`

But maybe include in the examples a little earlier. It is converted in the parameterization section.

**Comment by jrapin** [comment]
Hi @microprediction 
Where would you want to see it (= where have you been looking for it?)? And what would be a more intuitive syntax? (there are reasons for using this one, albeit maybe not good enough, so I am always curious to other options)
It does make sense to have it in the parametrization section since that's the point of the parametrization, then again I can cross-reference it somewhere else. Also, there are other ways to add constraints (through `set_cheap_constraints` with more flexibility but less robustness...  constraints are basically a whole work in progress.

**Comment by microprediction** [comment]
I'm guessing some would want to see the scipy syntax with [ (lower,upper), (lower,upper),... ] but your setup seems more flexible. 

No strong opinions but perhaps even the very first hello world example with 

`def square(x):
    return sum((x - 0.5) ** 2)`

could be illustrated on a bounded domain. After all, it is easier for readers to remove the bounds than add them. For instance I was slightly unsure as to whether I was supposed to use set_cheap_constraint in examples where (say) I'm just optimizing on a hypercube. Totally get that bounds are a work in progress.

**Comment by jrapin** [comment]
Hi @microprediction 
I just updated the doc  in the optimization section here https://facebookresearch.github.io/nevergrad/optimization.html#basic-example
Would that help? I'd rather not change the initial example since it would require the reader to understand the concept of the parametrization object, which is "hidden" in this case (using `parametrization=n` is a shortcut for `parametrization=ng.p.Array(shape=(n,))`).

Concerning scipy syntax, I indeed considered it for consistency (and kept the keywords), but I ruled it out so as to allow bounds to be broadcasted.
